### ISSUE TITLE
HTTP Grade D and E discriminate searx instances with a custom theme

### ISSUE THREAD

**Issue by unixfox** [dalf/searx-stats2#6]
Giving a lower rating (and probably ranking lower) because a searx instance customized the default theme is actually quite bad.
A few examples that customized the default theme and are apart from that perfectly good Searx instances:
- https://search.stinpriza.org/ - Changed only the logo
- https://suche.elaon.de/ - Changed only the logo
- https://spot.ecloud.global/ - Change the logo and customized the CSS and JS

And I could think of another example that would give a bad rating: minifying on the fly all the assets that aren't minifyied like for example [search_on_category_select.js](https://github.com/asciimoo/searx/blob/master/searx/static/plugins/js/search_on_category_select.js).

Actually the idea of giving grade based on the fact that the HTML files aren't completely the same as the original source code of Searx is a good idea but it's reaching a field where bad and good intentions are mixed together. And it's in this field, using the automated way to distinguish the bad from the good is actually very hard to do.

**Comment by dalf** [comment]
One idea to solve this is to use badges:
 - "Vanilla" : instance with unmodified content from the git repository.
 - "Unique" : instance with unique content.
 - "Approved": "Unique" content but approved (see below)
 - "External": URL to an another host.

Can be extended with:
 - "Google" - google analytics
 - "Matomo"
 - "Fathom"

Then the user can filter by these tags.

"approved" content can be stored in a git repository:
 - either by hash
 - either by URL (with something like wildcard)

Some examples:

File approved.yaml:
```yaml
- hash: ccea4aa40aca2ac30efc6cd85197886e0acc72fc73bb5695bb29ac73a3661616
  approved: true
  privacy-problem: false
  comment: xxxx
- hash: 4d3ada2f3687ff27df2184615e0e3ddb121388c7bac8a52af3bdabd6b4bdce6c
  approved: true
  privacy-problem: false
  comment: xxxx
```

File external.yaml:
```yaml
- url: https://*.cloudflare.com/*
  tags: cloudflare
  privacy-problem: true
- url: https://unpkg.com/*
  tags: cdn
- url: 	https://www.google-analytics.com/*
  tags: google
  privacy-problem: true
```

Then these files are modifed using PR which allow review and moderation.

... and most probably this is an overkill feature.

**Comment by unixfox** [comment]
I like this way without the "Approved" part because it's too much work to whitelist each resource one by one and each whitelisted resource could change at every possible moment.
Like for example if an instance decide to slightly change the logo this would render the whitelist incorrect for a few days/weeks.
I prefer the other way around, introducing a "Warning" badge. If someone has proof  that an instance is deliberately tracking his users trough the same host/domain he could open an issue/PR to add this badge.

About the external badge, we shouldn't make a list of possible host to whitelist or blacklist. External is like F, a Searx instance shouldn't contact external hosts. If the owner of the instance wants to customize his instance with a good intention, he just adds the needed resources in the source code.

**Comment by dalf** [comment]
@unixfox, is it the correct sum up  ?

The tags are about the static resources:

| Tag | Alias | Description|  Value set by |
|--|--|--|--|
| Vanilla | V (green) | instance with unmodified content from the searx git repository. | code: external_ressources.py |
| Customized | C (gray) | instance with modified content | code: external_ressources.py |
| External ressources | E (red) | URL to an another host. | code: external_ressources.py |
| Tracker | T (red) | the instance has a tracker | a human | 

Customized can be splited into two different tags:
* Customized JS (Cjs)
* Customized CSS & image (C)

The tag aliases would be displayed in the HTML column. When the pointer hovers a tag, the description and the related resources would be displayed.

----

Is "Tracker" the only tag set by someone ? Or is it an opened list defined by moderators ?

Example: an instance which modified the results to create affiliate links would have the tag "Customized". A tag "Affiliate links" can be added, but it is not more about the static resources.

**Comment by unixfox** [comment]
Why not combining trackers, ads, affiliate links and other dangerous/unwanted contents (like cryptomining) into a single tag: `Warning` or `Unsafe`? I think it's less a mess than having a tag for every kind of bad resources.

About the color of `Customized` I feel like again it's a bit discriminate them by giving a color very different from the `Vanilla` one. Actually the idea of making the initial color darker for the CSP could be also applied to the HTTP grade by giving a color mixing with green and gray like this hex color value: [#8aa15f](https://colorpalettes.net/color-palette-2289/).

The sorting for the HTTP grade should be by main color (green then red) so that both customized and vanilla Searx instances are treated equally.

Also, I like the idea of having two different type of customized instances: `C` and `Cjs`. Just like I said earlier we could apply the color darkening for that two types of tags by making `Cjs` slightly darker than `C`.

> The tag aliases would be displayed in the HTML column. When the pointer hovers a tag, the description and the related resources would be displayed.

:+1:

**Comment by dalf** [comment]
See https://searx.space/experimental/

**Comment by unixfox** [comment]
Thank you for your changes, I really like the new tags!

**Comment by dalf** [comment]
Fixed: #11
